### ISSUE TITLE
Adjustable Link Tile Density

### ISSUE THREAD

**Issue by bryanhunwardsen** [linkwarden/linkwarden#574]
**Is your feature request related to a problem? Please describe.**
I have 27, 32, 42, 55 inch 4K screens scaled to 200% in the OS, at 100% browser scale:
In View All Links: I can only see 8 link tiles + the top half of another 4. In list mode I can see 8.5 links with a lot of empty screen real estate. 

**Describe the solution you'd like**
Be able to choose tile density from presets or dynamically adjusted on a sliding scale in order to fit more on screen at one time.
Under the current design I can get all the pertinent information in a tile 1/3rd or 1/4th their current size, trippling or quadrupling then amount of date I can review before scrolling. 

**Describe alternatives you've considered**
Additional Feature requests to:
Collapse the Dashboard statistics pane into the empty header space between the search box and the account icon, freeing up nearly 1/3 of the screen for link tile display
Make left Nav pane collapible/minifiable and when not pinned flys out over the left sode links without impacting the tile layout to regain another column or two of visible links
Offer a 2 column list mode option
Have preset or user configurable titles so they can configure what information is important to them (i.e. favicon only, or favicon + name, favicon + base url, etc) having the titles ressize the the amount of detail the user has selected.

**Additional context**
An obfuscated screenshot of LW on my screens followed by some Microsoft Paint *cough terrible cough*  mockups of a layout that doesnt waste so much screen realestate with negative space/no content.

How it is now
![Obfuscated](https://github.com/linkwarden/linkwarden/assets/3408302/1f7fd895-b00a-49a9-86b6-87924ff21172)

Consolidating all the unused screen space
![Dense](https://github.com/linkwarden/linkwarden/assets/3408302/5f72b96b-9d46-4e73-9648-43960411affd)

Collection and Tag Pane flown off left side of screen
![Hidden](https://github.com/linkwarden/linkwarden/assets/3408302/0cda10e2-0f9c-4589-8307-2245d5ca1023)

Clicking or hovering of top left Dashboard icon causes Collection and Tags pane to fly over without rearranging the tiles until it loses focus or has action clicked on, make it pinnable that would then cause the link tiles to re-layout.
![fly over](https://github.com/linkwarden/linkwarden/assets/3408302/5d91b8e5-6a1a-4252-8d3e-553afed424ba)

**Comment by jcolinger3** [comment]
Agreed.  The usefulness of a helpful tool is diminished when the UI requires a TON of scrolling to find anything.

**Comment by skaboy71** [comment]
Fully Agree - This is needed.

**Comment by beancaper** [comment]
This + the ability to upload a custom icon/image for each link would make it much easier to identify links at a glance.

**Comment by daniel31x13** [comment]
[Done](https://github.com/linkwarden/linkwarden/releases/tag/v2.8.0) :)
